diff --git a/PrjBarlib/include/component.h b/PrjBarlib/include/component.h
index ace6bdd..bf32914 100644
--- a/PrjBarlib/include/component.h
+++ b/PrjBarlib/include/component.h
@@ -62,6 +62,11 @@ namespace bc
 			return resline->start;
 		}
 
+		Barscalar getCurrectLen()
+		{
+			return lastVal - resline->start;
+		}
+
 		Barscalar getLast()
 		{
 			return lastVal;
diff --git a/PrjBarlib/source/barcodeCreator.cpp b/PrjBarlib/source/barcodeCreator.cpp
index ff1bc3e..66bbd4d 100644
--- a/PrjBarlib/source/barcodeCreator.cpp
+++ b/PrjBarlib/source/barcodeCreator.cpp
@@ -96,7 +96,8 @@ 
 inline COMPP BarcodeCreator::attach(COMPP main, COMPP second)
 {
 	//second->kill();
-	if (second->getStart() == main->getStart() && main->getStart() == curbright)
+if ((second->getStart() == main->getStart() && main->getStart() == curbright) || 
+		(second->getCurrectLen() == 0 || main->getCurrectLen() == 0))
 	{
 #ifdef POINTS_ARE_AVAILABLE
 		for (const auto& val : second->resline->matr)
